src/extension.ts

```diff
@@ -300,24 +302,34 @@ async function canDelegateToJavaTestRunner(uri: vscode.Uri): Promise<boolean> {
     return (await vscode.commands.getCommands()).includes("java.test.editor.run");
 }
 
-async function launchTesting(uri: vscode.Uri, noDebug: boolean): Promise<void> {
-    noDebug ? vscode.commands.executeCommand("java.test.editor.run", uri) : vscode.commands.executeCommand("java.test.editor.debug", uri);
+function launchTesting(uri: vscode.Uri, noDebug: boolean, progressReporter: IProgressReporter) {
+    const command: string = noDebug ? "java.test.editor.run" : "java.test.editor.debug";
+    vscode.commands.executeCommand(command, uri, progressReporter);
+    if (compareVersions(getTestExtensionVersion(), "0.26.1") <= 0) {
```

Comment:
the run test from editor feature start from `0.25.0`.

So I think it should be

```js
if (compareVersions(getTestExtensionVersion(), "0.25.0") < 0) {
    throw new utility.OperationCancelledError("");
}
```

Reply:
Below is the execution flow when clicking run button:
the debugger pre-check the file info (show progress reporter)  -> trigger run command in test runner -> trigger configuration provider (show progress reporter)

If the test runner doesn't support progress reporter parameter at the second stage, the debugger needs to cancel the progress reporter at the stage 1, otherwise there is no way to terminate it.

so this is just used to check whether the test runner supports progress reporter.

Reply:
OK